src/components/Widget/inputs.tsx

```diff
@@ -17,6 +18,16 @@ export function useSyncWidgetInputs(defaultToken?: Currency) {
   const [tokens, setTokens] = useState<{ [Field.INPUT]?: Currency; [Field.OUTPUT]?: Currency }>({
     [Field.OUTPUT]: defaultToken,
   })
+
+  const previousDefaultToken = usePrevious(defaultToken)
+  useEffect(() => {
+    if (previousDefaultToken && previousDefaultToken !== defaultToken) {
```

Comment:
I don't think you need to check the previous token at all (and doing so adds extra state to the hook). Is there a case where you would _not_ want to update `tokens` when `defaultToken` updates? I'd just always reset it (if it's not already set):

    setTokens((tokens) => {
      if (tokens[Field.OUTPUT] === defaultToken) return tokens // this is a nop
      return { [Field.OUTPUT]: defaultToken }
    }, [defaultToken])

Reply:
ah i see, works for me

Reply:
probably don't even need the default token check inside actually?